Sources/PackageLoading/PackageBuilder.swift

```diff
@@ -268,6 +268,8 @@ public final class PackageBuilder {
         }
     }
 
+    private let outputFileName: (String, PackageModel.ProductType) -> RelativePath
```

Comment:
The package builder is the convention system that loads a package from disk, it doesn't need to care about the output file name. We should move the computation of output path/name into the class `ProductBuildDescription` (It is inside `BuildPlan.swift` file).